File: 08-Linear-search/linearSearch.java
import java.util.Scanner;

public class linearSearch {
    public static void main(String args[]) {
        System.out.println("Enter the size of the array:");
        Scanner sc = new Scanner(System.in);
        int size = sc.nextInt();
        int arr[] = new int[size];
        System.out.println("Enter the elements of the array:");
        for (int i = 0; i < size; i++) {
            arr[i] = sc.nextInt();
        }
        System.out.println("Your array is:");
        for (int i = 0; i < size; i++) {
            System.out.print(arr[i] + " ");
        }
        System.out.println("\nEnter the element you want to search:");
        Scanner s = new Scanner(System.in);
        int search = s.nextInt();
        for (int i = 0; i < size; i++) {
            if (arr[i] == search) {
                System.out.println("Element found at position " + i);
            }
        }
    }
}

File: C++/Searching/linear_search.cpp
#include <iostream>
using namespace std;

bool search(int arr[], int size, int key)
{
    for (int i = 0; i < size; i++)
    {
        if (arr[i] == key)
        {
            return 1;
        }
    }
    return 0;
}
int main()
{
    int arr[10] = {2, 7, 0, 1, -4, 9, 24, 19, -5, 11};
    int key;
    cout << "Enter the element you want to search: ";
    cin >> key;

    bool found = search(arr, 10, key);
    if (found)
    {
        cout << "Element is present";
    }
    else
    {
        cout << "Element is absent";
    }
}
